Sources/BSON/BSON.swift

```diff
@@ -302,6 +302,37 @@ extension BSON {
             return nil
         }
     }
+
+    /// Return this BSON as a `Double` if possible.
+    /// This will coerce numeric cases (e.g. `.decimal128`) into a `Double` if such coercion would be lossless.
+    public func toDouble() -> Double? {
```

Comment:
hmm yeah good point. this is just what was in the driver already. how do you think we should handle these?
for some conversions (e.g. Decimal128 to double) I think it's actually pretty straightforward because it looks like the Double initializer accepting a string will accept Decimal128 strings (e.g. "1.2E10" is accepted). but I don't believe the Int initializer accepts those, though I suppose we could maybe funnel the conversion through Double, like Int(Double(decimal128Str))?

alternatively we could just say these methods don't work with Decimal128s.

cc @nbbeeken

Reply:
Just giving it a whirl in the swift shell to see the results, this string is a valid Dec128 number
```swift
let d = Double("1.234567890123456789012345678901234E-7") 
// d: Double? = 1.2345678901234568E-7 
let i = Int(d!)
// i: Int = 0
```
So we could go with that method but it seems like loss of precision is not a typical use case for decimal 128, I'd vote for going with they don't work with d128

Reply:
You can avoid the lossy conversions by doing using the `Int(exactly:)` initializers. That being said, I think I agree that decimal128 probably shouldn't interface with these methods since it really is a different beast that needs to be handled carefully. For example, its definition of "lossy" is different than that of regular numbers, e.g. `BSONDecimal128("5") != BSONDecimal128("5.0")` (at least according to the existing tests).

That being said, we do have the existing `toDecimal128` method, so it may be confusing if certain numbers map _to_ decimal128 but decimal128 doesnt map back to them. I honestly am not really bothered by this, but I figured it was worth pointing out. I have a feeling that `toDecimal128` will be rarely if ever used in practice.

Reply:
Just to circle back on this--are we okay with just updating the docstring to remove the reference to decimal128?

Reply:
whoops, sorry I haven't had time to get back to this PR yet and make the changes around the default Codable conformance for BSONValues. that's fine with me though. did this come up elsewhere?

Reply:
Nope, just wanted to check if you were waiting on me for anything.

Reply:
updated the docstring.